### PR TITLE
CSS var: stepped process

### PR DESCRIPTION
**Pull request by louismaximepiton**
CSS var for stepped process

### CODE REVIEW

File: scss/_stepped-process.scss [hunk 1]
```diff
@@ -2,123 +2,141 @@
...
 .stepped-process-link {
-  flex: 1 0 $step-link-width;
-  max-width: $step-link-width;
+  flex: 1 0 var(--#{$prefix}stepped-process-link-width);
+  max-width: var(--#{$prefix}stepped-process-link-width);
   margin: auto;
   overflow: hidden;
-  line-height: 1;
-  color: $step-link-color;
+  line-height: 1; // not enough
```
Comment (louismaximepiton) [inline scss/_stepped-process.scss:94]: Needs to be tackled in another PR ?

**Review by louismaximepiton — COMMENTED** [review]
On hold until there's an answer for the second comment

File: scss/_stepped-process.scss [hunk 2]
```diff
@@ -2,123 +2,141 @@
...
+  --#{$prefix}stepped-process-item-next-bg: #{$step-item-next-bg};
+  --#{$prefix}stepped-process-item-drop-shadow: none;
+  --#{$prefix}stepped-process-item-arrow-width: 0;
+  --#{$prefix}stepped-process-item-arrow-shape: #{$step-item-arrow-shape};
+  --#{$prefix}stepped-process-link-width: #{$step-link-width};
+  --#{$prefix}stepped-process-link-color: #{$step-link-color};
+  --#{$prefix}stepped-process-link-active-color: #{$step-link-active-color};
+  --#{$prefix}stepped-process-link-marker: #{$step-link-marker};
+  --#{$prefix}stepped-process-link-text-decoration: #{$step-link-text-decoration};
+  // scss-docs-end stepped-process-css-vars
```
Comment (julien-deramond) [inline scss/_stepped-process.scss:24]: We didn't have the case before but shouldn't we include what's in `@include media-breakpoint-up` as well in the documentation inside some `scss-docs-end`? In the same as `stepped-process-css-vars` or in separate section explained in the docs.

File: scss/_stepped-process.scss (hunk 1 above)
Comment (julien-deramond) [reply scss/_stepped-process.scss:94]: Yes could you please create an issue about that and remove this comment?

File: scss/_variables.scss [hunk 3]
```diff
@@ -1967,26 +1967,29 @@ $back-to-top-icon-height:        subtract(1rem, 1px) !default;
...
+$stepped-process-counter:     step !default; // Used as a counter name
+
+$step-item-padding:       .5rem !default;
+$step-item-margin-end:    $border-width !default;
+$step-item-bg:            $black !default;
+$step-item-active-bg:     $primary !default;
+$step-item-next-bg:       $gray-400 !default;
+$step-item-shadow-size:   $border-width * 1.5 !default;
+$step-item-drop-shadow:   drop-shadow($step-item-shadow-size 0 0 $white) #{"/* rtl:"} drop-shadow(-$step-item-shadow-size 0 0 $white) #{"*/"} !default;
+// $step-item-padding-end:  $step-item-padding * 2 !default;
```
Comment (julien-deramond) [inline scss/_variables.scss:1982]: If not used anymore we should use `// fusv-disable` and `// fusv-enable` to handle the deprecation of the var in v5.2.0 (and then remove it afterwards)

File: scss/_variables.scss [hunk 4]
```diff
@@ -1967,26 +1967,29 @@ $back-to-top-icon-height:        subtract(1rem, 1px) !default;
...
+
+$step-item-arrow-width:   1rem !default;
+$step-item-arrow-shape:   polygon(0% 0%, subtract(100%, $border-width) 50%, 0% 100%) #{"/* rtl:"} polygon(100% 0%, $border-width 50%, 100% 100%) #{"*/"} !default; // Used in clip-path
+
+$step-link-width:         1.25ch !default; // Matches width of a single number
+$step-link-color:         $white !default;
+$step-link-active-color:  $black !default;
+$step-link-marker:        counter(var(--bs-stepped-process-counter)) inspect("\A0") !default;
+$step-link-marker-lg:     counter(var(--bs-stepped-process-counter)) inspect(".\A0") !default;
+$step-link-text-decoration: $link-decoration !default;
```
Comment (julien-deramond) [inline scss/_variables.scss:1992]: All `$step-*` and `$stepped-*` values should be vertically aligned in the file.

**Review by julien-deramond — CHANGES_REQUESTED** [review]
I haven't finished the technical review (CSS vars) but here are some first feedback.

I think the result observed in this PR is better but is it normal that the visible focus has changed? Not at all breakpoints but the shape is sometimes different (smaller).

File: scss/_stepped-process.scss (hunk 1 above)
Comment (louismaximepiton) [reply scss/_stepped-process.scss:94]: #1337

**Comment by julien-deramond** [comment]
@louismaximepiton Reverted the creation of `--#{$prefix}stepped-process-max-items` via https://github.com/Orange-OpenSource/Orange-Boosted-Bootstrap/pull/1281/commits/c94be7ba24d8132e1da58170a90e0c1659a5c602 because of `@for $i from 1 through $stepped-process-max-items {`. Indeed you won't be able to change the behavior of this loop by changing only the CSS var and having the value stored in this CSS var doesn't seem very useful in terms of reuse. We can discuss about that when you're back if I missed something and you want it to be reintroduced.

**Review by julien-deramond — APPROVED** [review]
LGTM! Visible focus has changed but will be revisited via https://github.com/Orange-OpenSource/Orange-Boosted-Bootstrap/issues/1189.
